internal/clients/http/gcp/gcp_client.go

```diff
@@ -0,0 +1,91 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/aws/smithy-go/ptr"
+	"github.com/rs/zerolog"
+)
+
+type gcpClient struct {
+	instancesClient *compute.InstancesClient
+	context         context.Context
+	logger          zerolog.Logger
+}
+
+func init() {
+	clients.GetGCPClient = newGCPClient
+}
+
+func newGCPClient(ctx context.Context) (clients.GCP, error) {
+	logger := ctxval.Logger(ctx).With().Str("client", "gcp").Logger()
+
+	instancesClient, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create new instances REST client: %w", err)
+	}
+
```

Comment:
This is missing `defer instancesClient.Close()`

Reply:
Ok after a small chat, I update: Close() must be part of the interface, make sure to document that all callers of GetGCPClient are responsible to defer the close call too. This is not HTTP, it's gRPC.